File: medium/3634-minimum-removals-to-balance-array/minimum-removals-to-balance-array.ts
function minRemoval(nums: number[], k: number): number {
    /**
        An array is considered balanced if the value of its maximum element is 
        at most k times the minimum element.

        I think this is straightforward.
     */


    nums.sort((a, b) => a - b);
    let n = nums.length;
    let i = 0;
    let maxWindow = 0;


    for (let j = 0; j < n; j++) {
        while (nums[j] > k * nums[i]) {
            i++;
        }

        maxWindow = Math.max(maxWindow, j - i + 1);
    }
    
    return n - maxWindow;
};

File: 3634-minimum-removals-to-balance-array/minimum-removals-to-balance-array.py
class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        nums.sort()
        n = len(nums)
        i = 0
        maxWindow = 0

        for j in range(n):
            while(nums[j] > k * nums[i]):
                i += 1
            maxWindow = max(maxWindow, j - i + 1)

        return n - maxWindow